File: Java/twosum.java
import java.util.HashMap;
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> MyMap=new HashMap<>();
        for(int i=0;i<nums.length;i++){
            int diff=target-nums[i];
            if(MyMap.containsKey(diff)){
                return new int[]{MyMap.get(diff),i};
            }
            MyMap.put(nums[i], i);
        }
        return new int[]{};
        
    }
}

File: python3/twosum.py
from typing import List
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        MyDict={}
        for i, n in enumerate(nums):
            diff=target-n
            if diff in MyDict:
                return [MyDict[diff], i]
            MyDict[n]=i
        